base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -1891,8 +1931,21 @@ fn handle_possible_reorg<T: BlockchainBackend>(
     chain_strength_comparer: &dyn ChainStrengthComparer,
     candidate_block: Arc<Block>,
 ) -> Result<BlockAddResult, ChainStorageError> {
```

Comment:
Maybe not related to this PR, but BlockAddREsult doesn't make sense here to return. For example, we return OrphanBlock when we are on the strongest chain